### PR TITLE
Direct call fixes for JITServer on Power

### PR DESCRIPTION
**Pull request by dchopra001**
We add `TR_MethodCallAddress` relocation types here and JITServer checks for direct calls. Additionally, we don't generate a relocation record when a function is calling itself (because we can generate a PC relative instruction back to the `startPC` instead).

### CODE REVIEW

File: runtime/compiler/p/codegen/J9PPCInstruction.cpp [hunk 1]
```diff
@@ -49,6 +49,7 @@ uint8_t *TR::PPCDepImmSymInstruction::generateBinaryEncoding()
       TR::ResolvedMethodSymbol *sym = getSymbolReference()->getSymbol()->getResolvedMethodSymbol();
       TR_ResolvedMethod *resolvedMethod = sym == NULL ? NULL : sym->getResolvedMethod();
       TR::LabelSymbol *label = getSymbolReference()->getSymbol()->getLabelSymbol();
+      bool myself = resolvedMethod != NULL && resolvedMethod->isSameMethod(comp->getCurrentMethod()) && !comp->isDLT();
```
Comment (ymanton) [inline runtime/compiler/p/codegen/J9PPCInstruction.cpp]:
```suggestion
      bool callToSelf = resolvedMethod != NULL && resolvedMethod->isSameMethod(comp->getCurrentMethod()) && !comp->isDLT();
```

File: runtime/compiler/p/codegen/J9PPCInstruction.cpp [hunk 2]
```diff
@@ -90,7 +91,7 @@ uint8_t *TR::PPCDepImmSymInstruction::generateBinaryEncoding()
             }
          }
 
-      if (resolvedMethod != NULL && resolvedMethod->isSameMethod(comp->getCurrentMethod()) && !comp->isDLT())
+      if (myself)
```
Comment (ymanton) [inline runtime/compiler/p/codegen/J9PPCInstruction.cpp]:
```suggestion
      if (callToSelf)
```

File: runtime/compiler/p/codegen/J9PPCInstruction.cpp [hunk 3]
```diff
@@ -135,10 +136,18 @@ uint8_t *TR::PPCDepImmSymInstruction::generateBinaryEncoding()
             }
          }
 
-      if (cg()->comp()->compileRelocatableCode() && label == NULL)
+      if ((cg()->comp()->compileRelocatableCode() || cg()->comp()->isOutOfProcessCompilation()) && label == NULL && !myself)
```
Comment (ymanton) [inline runtime/compiler/p/codegen/J9PPCInstruction.cpp]:
```suggestion
      if ((cg()->comp()->compileRelocatableCode() || cg()->comp()->isOutOfProcessCompilation()) &&
          label == NULL &&
          !callToSelf)
```

File: runtime/compiler/p/codegen/J9PPCInstruction.cpp [hunk 4]
```diff
@@ -135,10 +136,18 @@ uint8_t *TR::PPCDepImmSymInstruction::generateBinaryEncoding()
...
+         if (sym && !sym->isHelper() && resolvedMethod)
+	    {
+            cg()->addProjectSpecializedRelocation(cursor, (uint8_t *)getSymbolReference()->getMethodAddress(), NULL, TR_MethodCallAddress,
+                               __FILE__, __LINE__, getNode());
+	    }
+         else
+	    {
+            cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor,(uint8_t *)getSymbolReference(),TR_HelperAddress, cg()),
+                               __FILE__, __LINE__, getNode());
+	    }
```
Comment (ymanton) [inline runtime/compiler/p/codegen/J9PPCInstruction.cpp]: Indenting looks off in this block.

**Comment by dchopra001** [comment]
@ymanton All changes addressed here: e577eaf.

**Pull request merged**